### PR TITLE
Fix issue parsing a k:v pair element that starts with a keyword

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs [hunk 1]
```diff
@@ -18419,4 +18419,343 @@ void M()
             }
             """).VerifyDiagnostics();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75318")]
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs:18423]: >WorkItem("https://github.com/dotnet/roslyn/issues/75318") [](http://example.com/codeflow?start=11&length=57)

Is this issue related? If not, let's remove the `[WorkItem]` attribute from the added tests.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs]: done.

**Review by cston — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Review by RikkiGibson — APPROVED** [review]
